### PR TITLE
Move Event.slide to Slide.event

### LINKED ISSUE #2759: Make announcements depend on events instead of the other way around
### Describe the change
Make announcements depend on events instead of the other way around

### Motivation
Better modularity of apps

### Current implementation
Events can have a slide

### Suggested implementation
Slides can be about an event

### Additional context
Maybe make a class `EventAnnouncements` in a separate sub app for even better modularity

### PR DESCRIPTION
**Pull request by DeD1rk**
Closes #2759.

### Summary
This moves a ForeignKey to a OneToOne in the opposite direction.
After this, events only depends on announcements by including `SlideSerializer` in the event serializers.

### How to test
Steps to test the changes you made:
1. Check that the event screen still shows the right slide header if that was set before migrating.
2. Check that the events API still works.

### CODE REVIEW

**Comment by JobDoesburg** [comment]
> After this, events only depends on announcements by including SlideSerializer in the event serializers.

This is still problematic if you ask me...

**Comment by DeD1rk** [comment]
I mean I can simply remove it as it's not really used anyway in ThaliApp, but doing that is probably not possible for all of the inter-app-dependencies changes

File: website/announcements/templates/events/event.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% extends "events/event.html" %}
```
Comment (pingiun) [inline website/announcements/templates/events/event.html:1]: This makes the order of the apps important right? I think you should add a comment about that in the settings.py

**Review by pingiun — CHANGES_REQUESTED** [review]

File: website/announcements/templates/events/event.html (hunk 1 above)
Comment (DeD1rk) [reply website/announcements/templates/events/event.html:1]: That's right. I'll add a note. Once all of the unnecessary dependencies between apps are resolved, there should be a topological ordering or apps.

**Comment by pingiun** [comment]
The event admin should also still include the slide option, as event organisers probably expect to find it there

**Comment by JobDoesburg** [comment]
> The event admin should also still include the slide option, as event organisers probably expect to find it there

That could be implemented indeed (see also #2753), but I actually don't think event organizers think about this as the sliders are not really created by them, but by the promocie

**Comment by DeD1rk** [comment]
I added the inline. There will be a bit of uggliness though when also re-registering the EventAdmin from other apps (at least pushnotifications, but probably also food, sales, promotion): 

To subclass `EventAdmin`, we'd like to import it from `event.admin.event`. However, that version of the class stays the same even when re-registering one in announcements and/or pushnotifications. 

So if you try to override the original EventAdmin in two places, only one of the changes will stay. To prevent that, it looks like I'll have to e.g. check from `pushnotifications/admin` whether `announcements` is installed, and if so, import and extend `announcements.admin.EventAdmin`. Otherwise, import `events.admin.event.EventAdmin`.

That works but becomes quite confusing. Sadly the registry (`AdminSite._registry`) is not really exposed. I guess it should be possible to access it anyway (and take the currently registered class from there, rather than importing it) but that's also hacky.

File: website/events/api/v2/admin/serializers/event.py [hunk 2]
```diff
@@ -30,17 +28,13 @@ def to_internal_value(self, data):
         self.fields["organisers"] = serializers.PrimaryKeyRelatedField(
             queryset=MemberGroup.active_objects.all()
         )
-        self.fields["slide"] = serializers.PrimaryKeyRelatedField(
-            queryset=Slide.objects.all()
```
Comment (ColonelPhantom) [inline website/events/api/v2/admin/serializers/event.py:34]: Do we want to remove slides from the API? Or do we still want them, by fetching them from the onetoonekey?

Comment (JobDoesburg) [reply website/events/api/v2/admin/serializers/event.py:34]: I would propose removing them from the API if it doesn't break stuff

**Review by pingiun — APPROVED** [review]

**Pull request merged**